lib/dor/datastreams/content_metadata_ds.rb

```diff
@@ -35,10 +35,40 @@ class ContentMetadataDS < ActiveFedora::OmDatastream
 
     def public_xml
       result = ng_xml.clone
-      result.xpath('/contentMetadata/resource[not(file[(@deliver="yes" or @publish="yes")])]'   ).each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file[not(@deliver="yes" or @publish="yes")]'      ).each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file').xpath('@preserve|@shelve|@publish|@deliver').each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file/checksum').each { |n| n.remove }
+      
+      # remove any resources or attributes that are not destined for the public XML
+      result.xpath('/contentMetadata/resource[not(file[(@deliver="yes" or @publish="yes")]|externalFile)]').each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file[not(@deliver="yes" or @publish="yes")]'                ).each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file').xpath('@preserve|@shelve|@publish|@deliver'          ).each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file/checksum'                                              ).each { |n| n.remove }
+      
+      # support for dereferencing links via externalFile element(s) to the source (child) item - see JUMBO-19
+      result.xpath('/contentMetadata/resource/externalFile').each do |externalFile|
+        # enforce pre-conditions that resourceId, objectId, fileId are required
+        src_resource_id = externalFile['resourceId']
+        src_druid = externalFile['objectId']
+        src_file_id = externalFile['fileId']
+        fail ArgumentError, "Malformed externalFile data: #{externalFile.inspect}" if src_resource_id.nil? || src_file_id.nil? || src_druid.nil?
```

Comment:
Can any of these fields be ` ! x.nil?` but `x.empty?` ?  Could this check be `x.blank?` (maybe it's not available in dor-services, I think it's in a rails extension http://api.rubyonrails.org/v4.2.5/classes/Object.html#method-i-blank-3F).